File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
from collections import deque
def solution(arr):
    answer = []
    dq = deque()
    dq.append(arr[0])
    answer.append(arr[0])
    for i in arr:
        left = dq.popleft()
        if left == i:
            dq.append(i)
        else:
            dq.append(i)
            answer.append(i)
            
        
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int[] answer = {};
        
        Stack<Integer> st = new Stack<>();
        
        for(int num : arr){
            if(st.size() == 0){
                st.push(num);
                continue;
            }
            if(num == st.peek()){
                continue;
            }
            else{
                st.push(num);
            } 
        }
        int[] ret_int = new int[st.size()];
        int idx = 0;
        for(int s:st){
            ret_int[idx++] = s;
        }
        return ret_int;
    }
}